packages/shared/tsup.config.ts

```diff
@@ -1,44 +1,29 @@
 import type { Plugin } from 'esbuild';
 import { transform } from 'esbuild';
 import { readFile } from 'fs/promises';
-import type { Options } from 'tsup';
 import { defineConfig } from 'tsup';
 
-import { runAfterLast } from '../../scripts/utils';
 import { name, version } from './package.json';
 
 export default defineConfig(overrideOptions => {
   const isWatch = !!overrideOptions.watch;
-  const shouldPublish = !!overrideOptions.env?.publish;
 
-  const common: Options = {
-    entry: ['./src/**/*.{ts,tsx,js,jsx}'],
-    bundle: false,
+  return {
+    entry: ['./src/**/*.{ts,tsx}', '!./src/**/*.test.{ts,tsx}'],
+    format: ['cjs', 'esm'],
+    bundle: true,
```

Comment:
We need to have .mjs file extensions in imports of .mjs files - there's no way around it.

And for that we need to enable it, see https://github.com/egoist/tsup/issues/953

There's a papertrail of related issues for that:
- https://github.com/egoist/tsup/issues/565
- https://github.com/egoist/tsup/issues/760
- https://github.com/favware/esbuild-plugin-file-path-extensions
- https://github.com/evanw/esbuild/issues/622

Reply:
Have we considered the approach outlined [here](https://github.com/egoist/tsup/issues/953#issuecomment-1733903287)?

Reply:
On a separate but related note, `.mjs` extensions used to break CRA apps and this is the reason we didn't follow the ESM spec completely up until now and also the reason we used `legacyOutput` here. I will need time to find old support requests for this, but I'm torn if this is something we want to treat as a breaking change or not at this point. 

We need to discuss whether a build/bundle change on our side can be considered a breaking change. If the public API of our package has not changed, and the issue is caused by an outdated bundler on the user's side, I *think* we should consider this a *non* breaking change.

Reply:
> Have we considered the approach outlined https://github.com/egoist/tsup/issues/953#issuecomment-1733903287?

For that we'd need to add `type: "module"` to our `package.json` which could have larger implications. Because the workaround then renames CJS `.js` files to `.cjs` because `.js` will be treated as ESM by default then

Reply:
What I've seen from e.g. https://github.com/sindresorhus is that when they made packages ESM-only, it was a breaking change. Us adjusting the way how we output ESM here, I wouldn't consider that a breaking change.

Anyways, we're already publishing this as a new major, so it's fine?